src/commands/server/restore.ts

```diff
@@ -83,12 +114,193 @@ export default class Restore extends Command {
+            throw new Error(`Backup with name '${backupCrName}' is not successful`)
+          }
+          flags['backup-cr-name'] = backupCrName
+          task.title = `${task.title} ${backupCrName} found`
+        },
+      },
+      {
+        title: 'Gathering information about backup...',
+        enabled: () => flags['backup-cr-name'],
         task: async (_ctx: any, task: any) => {
-          const backupServerConfig = getBackupServerConfiguration(flags)
-          await requestRestore(flags.chenamespace, backupServerConfig, flags['snapshot-id'])
+          const backupCrName = flags['backup-cr-name']
+          const kube = new KubeHelper(flags)
+          const backupCr: V1CheClusterBackup | undefined = await kube.getCustomResource(flags.chenamespace, backupCrName, CHE_CLUSTER_API_GROUP, CHE_CLUSTER_API_VERSION, CHE_CLUSTER_BACKUP_KIND_PLURAL)
+          if (!backupCr) {
+            throw new Error(`Backup CR with name '${backupCrName}' not found`)
+          }
+
+          if (!backupCr.spec.backupServerConfigRef) {
+            throw new Error(`Backup CR '${backupCrName}' missing backup server configuration reference`)
+          }
+          if (!backupCr.status || !backupCr.status.cheVersion) {
+            throw new Error(`Backup CR '${backupCrName}' missing Che version`)
+          }
+          if (!backupCr.status || !backupCr.status.snapshotId) {
+            throw new Error(`Backup CR '${backupCrName}' missing snapshot ID`)
+          }
+
+          flags.version = backupCr.status.cheVersion
+          flags['snapshot-id'] = backupCr.status.snapshotId
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting backup server info...',
+        task: async (ctx: any, task: any) => {
+          // Get all information about backup server and validate it where possible
+          // before redeploying operator to requested version.
+          ctx.backupServerConfig = getBackupServerConfiguration(flags)
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting installation templates...',
+        enabled: (ctx: any) => flags.version && ctx.currentOperatorVersion !== flags.version && !flags.templates,
+        task: async (ctx: any, task: any) => {
+          if (flags.installer === 'olm') {
+            // Using embedded templates here as they are used in install flow for OLM.
+            // OLM install flow should be reworked to not to use templates, but OLM dependencies.
+            flags.templates = getEmbeddedTemplatesDirectory()
+          } else {
+            const getTemplatesTasks = new Listr(undefined, ctx.listrOptions)
+            getTemplatesTasks.add(checkChectlAndCheVersionCompatibility(flags))
+            getTemplatesTasks.add(downloadTemplates(flags))
+            return getTemplatesTasks
+          }
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Remove current Che operator',
```

Comment:
if `--rollback` is used we need to print restore info (version, snapshot id) and ask user's confirmation.

Reply:
ok